tests/test_connection_class.py

```diff
@@ -1475,6 +1475,71 @@ def test_read_wstring(self):
             )
             self.assertEqual(self.plc.read_by_name("wstr"), expected2)
 
+    def test_wstring_struct(self):
+        wstring_structure_def = (
+            ("name", pyads.PLCTYPE_WSTRING, 1),
+            ("value", pyads.PLCTYPE_INT, 1),
+        )
+
+        wstring_array_structure_def = (
+            ("name", pyads.PLCTYPE_WSTRING, 2),
+            ("value", pyads.PLCTYPE_INT, 1),
+        )
+
+        wstring_values = OrderedDict([
+            ("name", "foo bar"),
+            ("value", 24),
+        ])
+
+        wstring_array_values = OrderedDict([
+            ("name", ["foo bar", "Klaus Günter"]),
+            ("value", 24),
+        ])
+
+        # build structure value
+        data = "hällo world".encode("utf-16-le") + 2 * b"\x00"
+        byte_list = list(data) + (2 * (PLC_DEFAULT_STRING_SIZE + 1) - len(data)) * [0]
+        byte_list += [10, 0]
+        wstring_var = PLCVariable(
+            "wstring_struct",
+            value=bytes(byte_list),
+            ads_type=None,
+            symbol_type="S_WSTRING"
+        )
+        self.handler.add_variable(wstring_var)
+
+        data = "hällo world".encode("utf-16-le") + 2 * b"\x00"
+        byte_list = list(data) + (2 * (PLC_DEFAULT_STRING_SIZE + 1) - len(data)) * [0]
+        data = "foo bar".encode("utf-16-le") + 2 * b"\x00"
+        byte_list += list(data) + (2 * (PLC_DEFAULT_STRING_SIZE + 1) - len(data)) * [0]
+        byte_list += [10, 0]
+        wstring_array_var = PLCVariable(
+            "wstring_array_struct",
+            value=bytes(byte_list),
+            ads_type=None,
+            symbol_type="S_WSTRING_ARRAY"
+        )
+        self.handler.add_variable(wstring_array_var)
+
+        with self.plc:
+            # read WSTRING struct
+            val = self.plc.read_structure_by_name("wstring_struct", wstring_structure_def)
+            self.assertEqual({"name": "hällo world", "value": 10}, val)
+
+            # write WSTRING struct
+            self.plc.write_structure_by_name("wstring_struct", wstring_values, wstring_structure_def)
+            val = self.plc.read_structure_by_name("wstring_struct", wstring_structure_def)
+            self.assertEqual(wstring_values, val)
+
+            # read struct with WSTRING array
+            val = self.plc.read_structure_by_name("wstring_array_struct", wstring_array_structure_def)
+            self.assertEqual({"name": ["hällo world", "foo bar"], "value": 10}, val)
+
+            # write struct with WSTRING array
+            self.plc.write_structure_by_name("wstring_array_struct", wstring_array_values, wstring_array_structure_def)
+            val = self.plc.read_structure_by_name("wstring_array_struct", wstring_array_structure_def)
+            self.assertEqual(wstring_array_values, val)
+
 
```

Comment:
we should probably add some more tests for dict_to_bytes and bytes_to_dict to cover uses of WSTRING

Reply:
Also we should add tests with fixed STRING/WSTRING sizes.